core/shared/src/main/scala/magnolia.scala

```diff
@@ -278,17 +278,26 @@ object Magnolia {
         val preAssignments = caseParams.map(_.typeclass)
 
         val defaults = if (!isValueClass) {
-          val caseClassCompanion = genericType.companion
-          val constructorMethod = caseClassCompanion.decl(TermName("apply")).asMethod
-          val indexedConstructorParams =
-            constructorMethod.paramLists.head.map(_.asTerm).zipWithIndex
-
-          indexedConstructorParams.map {
-            case (p, idx) =>
-              if (p.isParamWithDefault) {
-                val method = TermName("apply$default$" + (idx + 1))
-                q"$scalaPkg.Some(${genericType.typeSymbol.companion.asTerm}.$method)"
-              } else q"$scalaPkg.None"
+          val constructorParams = genericType.decls.collect {
+            case a: MethodSymbol if a.isConstructor => a
+          }.head.paramLists.head.map(_.asTerm)
+          val noDefaults = constructorParams.forall(!_.isParamWithDefault)
+
+          if (noDefaults) {
```

Comment:
What is the purpose of the special case when none of the parameters have default values, rather than just letting the `else` block deal with each one in turn?

Reply:
When none of the parameters have default values, we can directly use the `constructorParameters` instead of having to get the `apply` method in the companion, which throws an exception when the case class is inside another class.

Do you know if it's possible to get the default parameter values without using the companion object?

Reply:
Ah, I see. I don't think it's easy. There might be a better way than the one I'm using, though... StackOverflow may have the answer... ;)